comps/animation/animation.py

```diff
@@ -0,0 +1,220 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import pathlib
+import platform
+import subprocess
+import time
+
+import cv2
+import ffmpeg
+import numpy as np
+
+# Wav2Lip-GFPGAN
+import requests
+import Wav2Lip.audio as audio
+import Wav2Lip.face_detection as face_detection
+from basicsr.utils import imwrite
+from gfpgan import GFPGANer
+from Wav2Lip.models import Wav2Lip
+
+cur_path = pathlib.Path(__file__).parent.resolve()
+comps_path = os.path.join(cur_path, "../")
+
+# Habana
+import habana_frameworks.torch.core as htcore
+import habana_frameworks.torch.hpu as hthpu
+from utils import *
+
+# GenAIComps
+from comps import (
+    AnimationDoc,
+    Base64ByteStrDoc,
+    ServiceType,
+    opea_microservices,
+    register_microservice,
+    register_statistics,
+    statistics_dict,
+)
+
+device = "hpu" if hthpu.is_available() else "cpu"
+print("Using {} for inference.".format(device))
+
+args = get_args()
+print("args: ", args)
+
+
+# Register the microservice
+@register_microservice(
+    name="opea_service@animation",
+    service_type=ServiceType.ANIMATION,
+    endpoint="/v1/animation",
+    host="0.0.0.0",
+    port=args.port,
+    input_datatype=Base64ByteStrDoc,
+)
+@register_statistics(names=["opea_service@animation"])
+def animate(input: Base64ByteStrDoc):
+    start = time.time()
+
+    if not os.path.exists("inputs"):
+        os.makedirs("inputs")
+    if not os.path.exists("temp"):
+        os.makedirs("temp")
+    if not os.path.exists("outputs"):
+        os.makedirs("outputs")
+
+    print(args.face, args.audio)
+    if not os.path.isfile(args.face):
+        raise ValueError("--face argument must be a valid path to video/image file")
+    elif args.face.split(".")[-1] in ["jpg", "jpeg", "png"]:
+        full_frames = [cv2.imread(args.face)]
+        fps = args.fps
+    else:
+        video_stream = cv2.VideoCapture(args.face)
```

Comment:
do you know where CV2 runs on? do they leverage Gaudi?  

Could we leverage Gaudi media pipeline?
https://docs.habana.ai/en/latest/Media_Pipeline/Media_Pipeline.html?highlight=media

Reply:
CV2 runs on cpu by default. If we want gpu support, we need to build CV2 from source. I don't see CV2 on Gaudi in the documentation.

Media pipeline is possible. Added as a TO-DO.